cpg-language-python/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/python/ExpressionHandler.kt

```diff
@@ -253,6 +253,34 @@ class ExpressionHandler(frontend: PythonLanguageFrontend) :
         }
     }
 
+    /**
+     * Joins the [nodes] with a [BinaryOperator] with the [operatorCode]. Nests the whole thing,
+     * where the first element in [nodes] is the lhs of the root of the tree of binary operators.
+     * The last operands are further down the tree.
+     */
+    fun joinListWithBinOp(
+        operatorCode: String,
+        nodes: List<Expression>,
+        rawNode: Python.AST.AST? = null,
+        isImplicit: Boolean = true
+    ): BinaryOperator {
+        val lastTwo = newBinaryOperator(operatorCode, rawNode = rawNode)
+        lastTwo.rhs = nodes.last()
+        lastTwo.lhs = nodes[nodes.size - 2]
+        return nodes.subList(0, nodes.size - 2).foldRight(lastTwo) { newVal, start ->
+            val nextValue = newBinaryOperator(operatorCode)
+            if (isImplicit && rawNode != null)
+                nextValue.implicit(
+                    code = frontend.codeOf(rawNode),
+                    location = frontend.locationOf(rawNode)
+                )
+            else if (isImplicit) nextValue.implicit()
```

Comment:
Yeah, this whole thing is quite useless. I set the `rawNode` now when generating the nodes. It's more an issue because we have like 5 ways which set the code and the location, each of which results in the exact same thing but by calling different methods...

Reply:
This is confusing me all the time. Is there any reason for having so many `implicit` functions???